_posts/namagic/2016-04-06-namagic-dictionary.md

```diff
@@ -0,0 +1,11 @@
+---
+author: namagic
+layout: post
+title: "More Dictionary Exercises"
+---
+
+<iframe src="https://trinket.io/embed/python3/5efacf1a6b" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+This activity was pretty hard. Outside of the first one, I actually couldn't get the rest to work. I frequently received error messages of KeyError which occurs when what I'm trying to grab out of the dictionary isn't actually in the dictionary. Figuring out which item I was looking for was also hard, but I was able to test it out even when I kept getting the KeyError because it would show me the word I attempted to select, but it couldn't print it because it wasn't actually in the dictionary. I think I'm getting confused as to what I'm actually parsing. 
```

Comment:
Go back over the code we used for counting.  It creates a key and sets it to something if the key isn't in the dictionary and if the key is in the dictionary it does something differently.  This is an idiom for creating historgam (counting) dictionaries.

When you're not counting, loop through a list and set the keys equal to something.